Functions/QuickScript/SteamScript/script.js

```diff
@@ -133,24 +108,16 @@ SteamScript.prototype.go = function() {
     // back to generic wait (might have been changed in preInstall)
     setupWizard.wait(tr("Please wait ..."));
 
-    // TODO enable "-silent" when progress bar works
-    // disabled it for now so the users can see the download progress
-    //wine.runInsidePrefix(wine.programFiles() + "/Steam/Steam.exe", ["-silent", "-applaunch", this._appId]);
     wine.runInsidePrefix(wine.programFiles() + "/Steam/Steam.exe", ["steam://install/" + this._appId]);
 
-    var bytesToDownload = this.getBytesToDownload(wine);
-    var bytesDownloaded = 0;
-    var progressBar = setupWizard.progressBar(tr("Please wait until Steam has finished the download ..."));
-    while (bytesDownloaded < bytesToDownload)
-    {
-        bytesDownloaded = this.getBytesDownloaded(wine);
-        progressBar.setProgressPercentage((bytesDownloaded / bytesToDownload) * 100);
-        progressBar.setText(tr("Downloaded {0} of {1} bytes", bytesDownloaded, bytesToDownload));
+    setupWizard.wait(tr("Please wait until Steam has finished the download ..."));
+
+    // wait until download started
+    while (!this.downloadStarted(wine)) {
         java.lang.Thread.sleep(100);
     }
 
-    // make sure download is really finished (download folder file size is not exact)
-    setupWizard.wait(tr("Please wait until Steam has finished the download ..."));
+    // make sure download is finished
     while (!this.downloadFinished(wine)) {
         java.lang.Thread.sleep(1000);
```

Comment:
I want to be sure that we do not get stuck here. If the download is really small, it might finish very soon. Therefore, this `sleep` is short. If we wait a bit too much to figure out if the download is finished, nothing bad happens. So we can sleep longer there to reduce the polling.